shop/internal/repository/repository_shop.go

```diff
@@ -0,0 +1,43 @@
+package repository
+
+import "github.com/GrabItYourself/giys-backend/lib/postgres/models"
+
+func (r *Repository) CreateShop(shop *models.Shop) error {
+	err := r.pg.Create(shop).Error
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Repository) GetShopById(id string) (*models.Shop, error) {
+	var shop models.Shop
+	err := r.pg.Where("id = ?", id).Take(&shop).Error
+	if err != nil {
+		return nil, err
+	}
+	return &shop, nil
+}
+
+func (r *Repository) EditShop(shop *models.Shop) (*models.Shop, error) {
+	err := r.pg.Model(&shop).Updates(shop).Error
+	if err != nil {
+		return nil, err
+	}
+	var edited_shop models.Shop
+	r.pg.Where("id = ?", shop.Id).Take(&edited_shop)
+	return &edited_shop, nil
+}
+
+func (r *Repository) DeleteShop(id string) error {
+	var shop models.Shop
+	err := r.pg.Where("id = ?", id).Take(&shop).Error
+	if err != nil {
+		return err
+	}
```

Comment:
do we have to find before delete tho? can't we just delete?

Reply:
If u want to delete with primary key, use [Delete with primary key](https://gorm.io/docs/delete.html#Delete-with-primary-key).

If you want to know if there really was a record that got deleted, use pg.Delete(...).RowsAffected
```go
result = r.pg.Delete(&models.Shop{}, id)
if result.Error != nil {
    return err
}
rowsAffected := result.RowsAffected
```

Reply:
If you want to return deleted shop, use [returning clause](https://gorm.io/docs/delete.html#Returning-Data-From-Deleted-Rows) similar to update.